File: examples/Python.ExampleApplication/Hooks/transformers/modify_permissions_transformer.py
from tableau_migration import (
    ContentTransformerBase,
    GranteeType,
    IPermissionSet
)

class ModifyPermissionsTransformer(ContentTransformerBase[IPermissionSet]):
    def transform(self, item_to_transform: IPermissionSet) -> IPermissionSet:
        filtered_grantees = [g for g in item_to_transform.grantee_capabilities if g.grantee_type != GranteeType.GROUP]
        return item_to_transform

File: examples/Csharp.ExampleApplication/Hooks/Transformers/ModifyPermissionsTransformer.cs
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.Extensions.Logging;
using Tableau.Migration.Content.Permissions;
using Tableau.Migration.Engine.Hooks.Transformers;
using Tableau.Migration.Resources;

namespace Csharp.ExampleApplication.Hooks.Transformers
{
    #region class
    public class ModifyPermissionsTransformer
        : ContentTransformerBase<IPermissionSet>
    {
        public ModifyPermissionsTransformer(ISharedResourcesLocalizer localizer, ILogger<IContentTransformer<IPermissionSet>> logger) 
            : base(localizer, logger)
        { }

        public override Task<IPermissionSet?> TransformAsync(IPermissionSet itemToTransform, CancellationToken cancel)
        {
            itemToTransform.GranteeCapabilities = itemToTransform.GranteeCapabilities
                .Where(g => g.GranteeType is GranteeType.Group)
                .ToList();

            return Task.FromResult<IPermissionSet?>(itemToTransform);
        }
    }
    #endregion
}
